spring-kafka/src/test/java/org/springframework/kafka/retrytopic/DeadLetterPublishingRecovererFactoryTests.java

```diff
@@ -62,6 +73,10 @@
 @SuppressWarnings({"unchecked", "rawtypes"})
 class DeadLetterPublishingRecovererFactoryTests {
 
+	private static final boolean isDebugEnabled =
+			((LogAccessor) ReflectionTestUtils.getField(DeadLetterPublishingRecovererFactory.class, "LOGGER"))
+					.isDebugEnabled(); // NOSONAR
```

Comment:
> Do we need to add some docs on the matter?

About that, I'm not sure we should add this to the docs until we fix the configuration system for RT's components - might create more confusion than the problem it solves. For most users the default setting might be good enough, and if we solve the configuration for `2.8.5` we can add that later.

But, of course, I can add it now too. WDYT?

Reply:
> This is something wrong to rely on the external static value which we don't control from this unit test.
> How about to use a `@LogLevels` on this class to be sure that our logging configuration is explicit for the test logic?

Oh, I didn't know about that annotation, cool! I'll use that then, thanks.

Reply:
OK. Just fix the test and `@since` and I'll merge.

As I said on other your PR, the RT configuration refactoring is too drastic to be back-ported even to `2.8.x`.
It works some way over there already, so that's enough. 
The new vision or hard refactoring is better to do in new version (I guess `2.9.0` is good enough). So, it is easy for you to implement the feature without a breaking changes fear and it is more comfortable for end-users to refactor their projects to new API from our new version in their own new versions.

Reply:
Well, about the configuration, I thought we came to the conclusion that if we can come up with something that is an optional add-on that that users can just not use if there's a problem or they simply don't want or need to, we might merge it to 2.8.5. But I think we'll have a clearer picture when I have something more concrete we can look into.

TBH, I think the current configuration system is too bad, specially when we consider that the next version in sync with Boot might be 3.0.0 at end of year. So if we decide not to back port this new approach, I'll still try to come up with something better for 2.8.x, even if it's only for the main features - of course, if that's not a problem.

Reply:
I seem to be much more worried about this than you 😄 Maybe I'm overreacting. Probably @garyrussell can help settling this since he's been involved with this feature from the beginning. And we sure can take a better look after I work on the `Support`-based configuration.

But to provide some more context, we didn't use to have many features that relied on this kind of configuration - most features would use the `@RetryableTopic` annotation and `RetryTopicConfigurationBuilder`, which are APIs thought from the start to be used by end-users. That's probably why we didn't have so much bad feedback on this.

In the past couple versions, we've introduced a few features - e.g. the ability to set [framework level](https://docs.spring.io/spring-kafka/reference/html/#retry-topic-ex-classifier) `FATAL` exceptions, and the ability to [combine blocking and non-blocking retries](https://docs.spring.io/spring-kafka/reference/html/#retry-topic-combine-blocking), that can only be used through this kind of configuration.

Those seem to be gathering user attention - I think it took a couple of days from Boot launching with this last feature and the issue popping on SO.

So, as you've probably already have seen, to configure that feature, currently users might have to do:

```java
@Configuration
    public static class SO71705876Configuration {

        @Bean(name = RetryTopicInternalBeanNames.LISTENER_CONTAINER_FACTORY_CONFIGURER_NAME)
        public ListenerContainerFactoryConfigurer lcfc(KafkaConsumerBackoffManager kafkaConsumerBackoffManager,
                DeadLetterPublishingRecovererFactory deadLetterPublishingRecovererFactory) {
            ListenerContainerFactoryConfigurer lcfc = new ListenerContainerFactoryConfigurer(kafkaConsumerBackoffManager, deadLetterPublishingRecovererFactory, Clock.systemUTC());
            lcfc.setBlockingRetryableExceptions(IllegalArgumentException.class, IllegalStateException.class);
            lcfc.setBlockingRetriesBackOff(new FixedBackOff(500, 5)); // Optional
            return lcfc;
        }

        @Bean(name = RetryTopicInternalBeanNames.KAFKA_CONSUMER_BACKOFF_MANAGER)
        public KafkaConsumerBackoffManager backOffManager(ApplicationContext context) {
            PartitionPausingBackOffManagerFactory managerFactory =
                    new PartitionPausingBackOffManagerFactory();
            managerFactory.setApplicationContext(context);
            return managerFactory.create();
        }

        @Bean(name = RetryTopicInternalBeanNames.DEAD_LETTER_PUBLISHING_RECOVERER_FACTORY_BEAN_NAME)
        public DeadLetterPublishingRecovererFactory dlprFactory(DestinationTopicResolver resolver) {
            var DLPRF = new DeadLetterPublishingRecovererFactory(resolver);
            dlprf.neverLogListenerException();
            return new DeadLetterPublishingRecovererFactory(resolver);
        }

        @Bean(name = RetryTopicInternalBeanNames.DESTINATION_TOPIC_CONTAINER_NAME)
        public DestinationTopicResolver destinationTopicResolver(ApplicationContext context) {
            return new DefaultDestinationTopicResolver(Clock.systemUTC(), context);
        }
```

I think it was bad enough when we thought users had do declare only one bean in such way, but I think having to do all this for any feature is really not acceptable in my book. And it tends to get worse as we release new features - albeit we could defer some new things to `2.9.0`. But the logging configuration from the other PR is an example of this.

Let alone users having to look into `SO` for this workaround after being frustrated by errors - and I'd feel very bad if we actually documented this ugliness as an "official" workaround 😄 

So that's to provide some more context. Again, maybe I'm overreacting. I should have something on the `Support` configuration by Monday so we can take a better look.

I really appreciate you looking into all this. Thanks!

Reply:
I think you are hijacking this PR for other matter 😄 
Perhaps would be great if you are linking to this comment from the proper place.

Thanks